From be2b920ead2ccdd0435b3a0029c916c021c8df44 Mon Sep 17 00:00:00 2001
From: Tom White <tom@cloudera.com>
Date: Fri, 31 Aug 2012 10:39:53 +0100
Subject: [PATCH 132/795] MR1: MAPREDUCE-4610. Support deprecated mapreduce.job.counters.limit property in MR2.

Reason: Compatibility
Ref: CDH-7678
Author: Tom White
(cherry picked from commit 36442f5f8b2f2aa1044a1f21740a792698b7e218)
(cherry picked from commit db51009db4b04fbed0d76a17aa799b6d5936a7dc)
---
 .../apache/hadoop/mapreduce/counters/Limits.java   |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapreduce/counters/Limits.java b/src/mapred/org/apache/hadoop/mapreduce/counters/Limits.java
index d22ac70..597b01b 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/counters/Limits.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/counters/Limits.java
@@ -32,8 +32,11 @@
       conf.getInt(COUNTER_NAME_MAX_KEY, COUNTER_NAME_MAX_DEFAULT);
   public static final int GROUPS_MAX =
       conf.getInt(COUNTER_GROUPS_MAX_KEY, COUNTER_GROUPS_MAX_DEFAULT);
+  // mapreduce.job.counters.limit is deprecated in favor of
+  // mapreduce.job.counters.max in Hadoop 2 so we support both here
   public static final int COUNTERS_MAX =
-      conf.getInt(COUNTERS_MAX_KEY, COUNTERS_MAX_DEFAULT);
+      conf.getInt(COUNTERS_MAX_KEY,
+        conf.getInt("mapreduce.job.counters.limit", COUNTERS_MAX_DEFAULT));
 
   private int totalCounters;
   private LimitExceededException firstViolation;
-- 
1.7.0.4

